src/hotspot/share/utilities/elfFile.hpp

```diff
@@ -897,6 +944,18 @@ class DwarfFile : public ElfFile {
    *  More details about the different phases can be found at the associated methods.
    */
   bool get_filename_and_line_number(uint32_t offset_in_library, char* filename, size_t filename_len, int* line, bool is_pc_after_call);
+
+ private:
+  ArangesCache _aranges_cache;
+
+  // Build cache of all address ranges for binary search in a single pass
+  void ensure_aranges_cache() {
+    if (_aranges_cache._initialized || _aranges_cache._failed) {
+      return;
+    }
+    DebugAranges debug_aranges(const_cast<DwarfFile*>(this));
```

Comment:
I don't think the `const_cast` is required since the constructor takes a non-const pointer.


```suggestion
    DebugAranges debug_aranges(this);
```